wagtail/contrib/sitemaps/sitemap_generator.py

```diff
@@ -1,13 +1,19 @@
+import warnings
+
 from django.contrib.sitemaps import Sitemap as DjangoSitemap
 
+from wagtail.core.utils import accepts_kwarg
+from wagtail.utils.deprecation import RemovedInWagtail22Warning
+
 
 class Sitemap(DjangoSitemap):
 
-    def __init__(self, site=None):
+    def __init__(self, site=None, request=None):
         self.site = site
+        self.request = request
```

Comment:
In all uses done of `Sitemap`, the site passed is `request.site`. So we actually don’t need to pass `site` anymore.